sklearn/tests/test_common.py

```diff
@@ -219,3 +229,156 @@ def test_get_params_invariance():
+            for _ in range(weight):
+                aug_data.append(samples)
+        aug_data = map(np.array, zip(*aug_data))
+        return aug_data
+
+    train, test = train_test_split(range(n_samples))
+
+    for name, Estimator in estimators:
+
+        if name in exclude:
+            print ("%s is being excluded" % name)
+            continue
+        if not has_fit_parameter(Estimator, 'sample_weight'):
+            continue
+        if is_classifier(Estimator):
+            X, y = X_clf, y_clf
+        elif is_regressor(Estimator):
+            X, y = X_reg, y_reg
+        else:
+            print ("%s is neither classifier nor regressor" % name)
+            continue
+
+        estimator_sw = Estimator()
+        set_random_state(estimator_sw, random_state=random_state)
+        estimator_sw.fit(X[train], y[train], sample_weight=sample_weight[train])
+        X_aug_train, y_aug_train = aug((X[train], y[train]),
+                                   sample_weight[train])
+        assert_equal(X_aug_train.shape[0], np.sum(sample_weight[train]))
+
+        estimator_aug = Estimator()
+        set_random_state(estimator_aug, random_state=random_state)
+        estimator_aug.fit(X_aug_train, y_aug_train)
+
+        precision = 6
+        # if estimator has `coef_` attribute, then compare the two
+        if hasattr(estimator_sw, 'coef_'):
+            yield (assert_array_almost_equal,
+                   estimator_sw.coef_, estimator_aug.coef_, precision, name+' coef_ not equal')
+
+        pred_sw = estimator_sw.predict(X[test])
+        pred_aug = estimator_aug.predict(X[test])
+
+        yield assert_array_almost_equal, pred_sw, pred_aug, precision, name+' prediction not equal'
+
+
+
+def test_sample_weight_0(random_state=42):
+    exclude = [
+               'AdaBoostClassifier', 'AdaBoostRegressor',
+               'BaggingClassifier', 'BaggingRegressor',
+               'GradientBoostingClassifier', 'GradientBoostingRegressor',
+               'ExtraTreeClassifier', 'ExtraTreeRegressor',
+               'ExtraTreesClassifier', 'ExtraTreesRegressor',
+               'DecisionTreeClassifier','DecisionTreeRegressor',
+               'LogisticRegression', 'LogisticRegressionCV', 'LinearSVC',
+               'MultinomialNB',  # Requires positive samples
+               'CalibratedClassifierCV',  # This is a meta-estimator using LinearSVC
+               'SGDClassifier',  # Doesn't work. Probably more data needed
+               'SGDRegressor',  # Doesn't work. Probably more data needed
+               'Perceptron',  # Uses SGD too. Doesn't work. Probably more data needed
+               'RidgeClassifierCV', 'RidgeCV',
+               'RandomForestClassifier', 'RandomForestRegressor',
+               ]
+    estimators = all_estimators()
+
+    n_samples, n_features = 20, 5
+    rng = check_random_state(random_state)
+
+    sample_weight = (rng.permutation(n_samples) < (n_samples / 2.)).astype(np.float)
```

Comment:
I was not suggesting to do that. I think `int` makes more sense in this case.
My argument is that your tests will not cover the situation in which learning algorithms are weighted with `float` sample weights, as there is not a corresponding interpretation of "weight = number of sample copy". I simply do not think there is a way to cover this situation though.